File: doc-examples/example-jsonb-java/src/main/java/example/LowerPriorityColorSerializer.java
package example;

import io.micronaut.context.annotation.Requires;
import jakarta.annotation.Priority;
import jakarta.inject.Singleton;
import jakarta.json.bind.serializer.JsonbSerializer;
import jakarta.json.bind.serializer.SerializationContext;
import jakarta.json.stream.JsonGenerator;

@Singleton
@Requires(property = "spec.name", value = "jsonb-extension-beans")
@Priority(20)
public final class LowerPriorityColorSerializer implements JsonbSerializer<Color> {
    @Override
    public void serialize(Color obj, JsonGenerator generator, SerializationContext ctx) {
        generator.write("fallback-" + obj.getValue());
    }
}

File: doc-examples/example-jsonb-python/src/main/python/example/LowerPriorityColorSerializer.py
from jakarta.annotation import Priority
from jakarta.inject import Singleton
from jakarta.json.bind.serializer import JsonbSerializer, SerializationContext
from jakarta.json.stream import JsonGenerator
from micronaut.context.annotation import Requires

from example.Color import Color


@Singleton
@Requires(property="spec.name", value="jsonb-extension-beans")
@Priority(20)
class LowerPriorityColorSerializer(JsonbSerializer[Color]):
    def serialize(self, obj: Color, generator: JsonGenerator, ctx: SerializationContext) -> None:
        generator.write("fallback-" + obj.value)
